core/src/main/scala/org/apache/spark/scheduler/TaskSetManager.scala

```diff
@@ -799,7 +799,7 @@ private[spark] class TaskSetManager(
     if (successful(index)) {
       logInfo(
         s"Task ${info.id} in stage ${taskSet.id} (TID $tid) failed, " +
-        "but another instance of the task has already succeeded, " +
+        "but it is already marked as successful, " +
```

Comment:
How about 

"Task ${info.id} in stage ${taskSet.id} (TID $tid) failed, but the task will not be re-executed (either because the task failed with a fetch failure, so the previous stage needs to be re-run, or because a different copy of the task has already succeeded)"